src/compute/src/render/context.rs

```diff
@@ -91,16 +95,19 @@ where
     S::Timestamp: Lattice + Refines<mz_repr::Timestamp>,
 {
     /// Creates a new empty Context.
-    pub fn for_dataflow<Plan>(
+    pub fn for_dataflow_in<Plan>(
         dataflow: &DataflowDescription<Plan, CollectionMetadata>,
-        dataflow_id: usize,
+        scope: S,
     ) -> Self {
+        use mz_ore::collections::CollectionExt as IteratorExt;
+        let dataflow_id = scope.addr().into_element();
```

Comment:
This line is getting the tests to fail because we provide the `region` nested in the top-level `scope` in the calls to this function. I'd suggest we make it instead:
```suggestion
        let dataflow_id = scope.addr().into_first();
```
That way, we would always get the first address component for the top-level scope.